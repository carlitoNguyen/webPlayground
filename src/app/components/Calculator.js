/* eslint-disable no-unused-vars */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

class Calculator extends React.Component {
	constructor(props) {
		super(props);
	}

	render() {
		return (
			<div className="container">
				<label className="content-header-label">Calculation</label>
				<div className="content-header info">{this.props.operation}</div>
				<label className="content-header-label">Result</label>
				<div className="content-header success">{this.props.result}</div>
				<div className="content">
					<div className="content-center">
						<div>
							<button onClick={() => this.props.onUpdate('1')}>1</button>
							<button onClick={() => this.props.onUpdate('2')}>2</button>
							<button onClick={() => this.props.onUpdate('3')}>3</button>
						</div>
						<div>
							<button onClick={() => this.props.onUpdate('4')}>4</button>
							<button onClick={() => this.props.onUpdate('5')}>5</button>
							<button onClick={() => this.props.onUpdate('6')}>6</button>
						</div>
						<div>
							<button onClick={() => this.props.onUpdate('7')}>7</button>
							<button onClick={() => this.props.onUpdate('8')}>8</button>
							<button onClick={() => this.props.onUpdate('9')}>9</button>
						</div>
						<div>
							<button className="clear" onClick={this.props.onReset}>C</button>
							<button onClick={() => this.props.onUpdate('0')}>0</button>
							<button onClick={() => this.props.onUpdate('.')}>.</button>
						</div>
					</div>
					<div className="content-left">
						<button onClick={() => this.props.onUpdate('+')}>+</button>
						<button onClick={() => this.props.onUpdate('-')}>-</button>
						<button onClick={() => this.props.onUpdate('/')}>/</button>
						<button onClick={() => this.props.onUpdate('*')}>x</button>
					</div>
				</div>
				<div className="content-footer">
					<button onClick={this.props.onSubmit}>=</button>
				</div>
			</div>
		);
	}
}

Calculator.propTypes = {
	onSubmit: PropTypes.func,
	onReset: PropTypes.func,
	onUpdate: PropTypes.func,
	operation: PropTypes.string,
};

const mapStateToProps = state => ({
	operation: state.operation,
	result: state.result,
});

const mapDispatchToProps = dispatch => ({
	onSubmit: () => dispatch({ type: 'SUBMIT'}),
	onReset: () => dispatch({ type: 'RESET'}),
	onUpdate: (data) => dispatch({ type: 'UPDATE', data}),
});

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(Calculator)
