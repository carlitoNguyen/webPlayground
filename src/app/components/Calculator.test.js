import React from 'react';
import Calculator from './Calculator';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux'

describe('Connected Calculator',()=> {
	const initialState = { result: 10, operation: "7+3" };
	const mockStore = configureStore();

	it('render a calculator', () => {
		const store = mockStore(initialState);
		const wrapper = mount(<Provider store={store}><Calculator/></Provider>);
		expect(wrapper).toMatchSnapshot();
	});
});
