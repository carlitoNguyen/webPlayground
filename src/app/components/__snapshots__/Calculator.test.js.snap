// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Calculator render a calculator 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Calculator)>
    <Calculator
      onReset={[Function]}
      onSubmit={[Function]}
      onUpdate={[Function]}
      operation="7+3"
      result={10}
    >
      <div
        className="container"
      >
        <label
          className="content-header-label"
        >
          Calculation
        </label>
        <div
          className="content-header info"
        >
          7+3
        </div>
        <label
          className="content-header-label"
        >
          Result
        </label>
        <div
          className="content-header success"
        >
          10
        </div>
        <div
          className="content"
        >
          <div
            className="content-center"
          >
            <div>
              <button
                onClick={[Function]}
              >
                1
              </button>
              <button
                onClick={[Function]}
              >
                2
              </button>
              <button
                onClick={[Function]}
              >
                3
              </button>
            </div>
            <div>
              <button
                onClick={[Function]}
              >
                4
              </button>
              <button
                onClick={[Function]}
              >
                5
              </button>
              <button
                onClick={[Function]}
              >
                6
              </button>
            </div>
            <div>
              <button
                onClick={[Function]}
              >
                7
              </button>
              <button
                onClick={[Function]}
              >
                8
              </button>
              <button
                onClick={[Function]}
              >
                9
              </button>
            </div>
            <div>
              <button
                className="clear"
                onClick={[Function]}
              >
                C
              </button>
              <button
                onClick={[Function]}
              >
                0
              </button>
              <button
                onClick={[Function]}
              >
                .
              </button>
            </div>
          </div>
          <div
            className="content-left"
          >
            <button
              onClick={[Function]}
            >
              +
            </button>
            <button
              onClick={[Function]}
            >
              -
            </button>
            <button
              onClick={[Function]}
            >
              /
            </button>
            <button
              onClick={[Function]}
            >
              x
            </button>
          </div>
        </div>
        <div
          className="content-footer"
        >
          <button
            onClick={[Function]}
          >
            =
          </button>
        </div>
      </div>
    </Calculator>
  </Connect(Calculator)>
</Provider>
`;
